server/service/invites.go

```diff
@@ -0,0 +1,65 @@
+package service
+
+import (
+	"context"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+////////////////////////////////////////////////////////////////////////////////
+// Update invite
+////////////////////////////////////////////////////////////////////////////////
+
+type updateInviteRequest struct {
+	ID uint `url:"id"`
+	fleet.InvitePayload
+}
+
+type updateInviteResponse struct {
+	Invite *fleet.Invite `json:"invite"`
+	Err    error         `json:"error,omitempty"`
+}
+
+func (r updateInviteResponse) error() error { return r.Err }
+
+func updateInviteEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
+	req := request.(*updateInviteRequest)
+	invite, err := svc.UpdateInvite(ctx, req.ID, req.InvitePayload)
+	if err != nil {
+		return updateInviteResponse{Err: err}, nil
+	}
+
+	return updateInviteResponse{Invite: invite}, nil
+}
+
+func (svc *Service) UpdateInvite(ctx context.Context, id uint, payload fleet.InvitePayload) (*fleet.Invite, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Invite{}, fleet.ActionWrite); err != nil {
```

Comment:
For editing queries: it first checks for read access to "(all) queries", then reads the `Query` from the db, then checks for write access to the specified query, then proceeds to write. Any reason we are using that approach there? (vs like here where it only checks for write at the beginning)

Reply:
In this case, invite write is a permission that is "bigger" than invite read. So if you can do that, you can read them.